### DISCUSSION TITLE
How to repeat the same file within the table of content _toc file of jupyterbook

### DISCUSSION THREAD

**Discussion by marsup13** [executablebooks/meta#931] [General]
Hello, 

I was wondering please is there a way to repeate the same file (ipynb or markdown ) in the table of content _toc file of jupyterbook project without having errors or needing to duplicate the files and rename them differently (I want to include the same file content in multiple places/chapters of the generated pdf/html book)?

Is there a way to have something similar to the next (where I am repeating file1, file2 and file3 for the three chapters)?

# Table of contents
# Learn more at https://jupyterbook.org/customize/toc.html

format: jb-book
root: intro
parts:
  - caption: CAPTION1
    chapters:
    - file: file1
    - file: file2
    - file: file3
    - file: file4
    - file: file5
    - file: file6
  

  - caption: CAPTION2
    chapters:
    - file: file1
    - file: file2
    - file: file3
    - file: file8
    - file: file9
    - file: file16

  - caption: CAPTION3
    chapters:
    - file: file1
    - file: file2
    - file: file3
    - file: file14
    - file: file25
    - file: file26

Thanks and have a great day!

**Comment by choldgraf** [comment]
### Your question

Hmmm, I think the best way that you could do this is via an `{include}` directive. You can't put the same file multiple places in the TOC without a warning. But you could:

- Put the content you wish to replicate _outside_ of your TOC (so it is excluded)
- In many places in your docs, use:
   
   ````
   ```{include} path/to/yourfile.ipynb
   ```
   ````

### [meta] On creating discussions

Oops - I converted the issue you created from this to a discussion because this is better as a discussion here and not an issue, and then realized you'd already opened up a discussion. So I'll delete this.

For general Q&A, please keep them to discussions, not issues - we use issues here to track things to do for the contributing team.

**Reply by marsup13** [reply]
Thanks for the clarifications @choldgraf .

Moreover, I have a another question.

Since I didn't receive a quick answer, the workaround I have tried was to put the repeated files into different folders and then include them in the _toc.yml.

The problem when doing such thing, the build of the jupyterbook project was also failed .

The build output tells me to look at the logs of sphinx (I am using pycharm, a pycharm project on windows 11 using Ubuntu WSL, python 3.8.10, pip environment).

After looking at the logs, the next error keeps showing up (even after many trials to change the paths for the different used files, I even put the full path of the files, and the build process still can't recognize my files situated at the different folders)

````                                                                                              
    self._write_serial(sorted(docnames))
  File "/home/ma/python-environments/report/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 531, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/home/ma/python-environments/report/lib/python3.8/site-packages/sphinx/environment/__init__.py", line 535, in get_and_resolve_doctree
    toctreenode.replace_self([])
  File "/home/ma/python-environments/report/lib/python3.8/site-packages/docutils/nodes.py", line 1034, in replace_self
    assert not self[att], \
AssertionError: Losing "classes" attribute: latex-tableofcontents
````


I am using a _toc structure like the next:


format: jb-book
root: index
parts:
  - caption: Name of Part 1
    chapters:
    - file: path/to/part1/chapter1
    - file: path/to/part1/chapter2
      sections:
      - file: path/to/part1/chapter2/section1
  - caption: Name of Part 2
    chapters:
    - file: path/to/part2/chapter1
    - file: path/to/part2/chapter2
      sections:
      - file: path/to/part2/chapter2/section1
